In [1]:
import pandas as pd

from lifelines import CoxPHFitter

Read the data

In [2]:
DatOriginal = pd.read_csv("DatasetsMedicalResearch/Survival times of patients with melanoma.dat", sep="\s+")

<>:1: SyntaxWarning: invalid escape sequence '\s'
<>:1: SyntaxWarning: invalid escape sequence '\s'
/tmp/ipykernel_6113/3111034151.py:1: SyntaxWarning: invalid escape sequence '\s'
  DatOriginal = pd.read_csv("DatasetsMedicalResearch/Survival times of patients with melanoma.dat", sep="\s+")


In [3]:
DatOriginal.head()

age  treatment  time  status
0    1          1    19       1
1    1          1    24       0
2    1          1     8       1
3    1          1    17       0
4    1          1    17       0

The data has two treatments and three strata in the age.

In [4]:
cph = CoxPHFitter()
cph.fit(DatOriginal, 'time', 'status', formula='C(treatment)', strata='age')
cph.print_summary()

<lifelines.CoxPHFitter: fitted with 30 total observations, 20 right-censored observations>
             duration col = 'time'
                event col = 'status'
                   strata = age
      baseline estimation = breslow
   number of observations = 30
number of events observed = 10
   partial log-likelihood = -19.35
         time fit was run = 2024-09-07 06:40:58 UTC

---
                   coef exp(coef)  se(coef)  coef lower 95%  coef upper 95% exp(coef) lower 95% exp(coef) upper 95%
covariate                                                                                                          
C(treatment)[T.2] -0.53      0.59      0.65           -1.79            0.74                0.17                2.09

                   cmp to     z    p  -log2(p)
covariate                                     
C(treatment)[T.2]    0.00 -0.82 0.41      1.27
---
Concordance = 0.54
Partial AIC = 40.71
log-likelihood ratio test = 0.66 on 1 df
-log2(p) of ll-ratio test = 1.27

Because the $P$-value is 0.41 we conclude that there is no significance difference between survival times for the two treatments